website/content/v1.8/advanced/overlays.md

```diff
@@ -43,3 +43,50 @@ An Overlay is a container image with the [specific folder structure](https://git
 Overlays can be built and managed using any tool that produces container images, e.g. `docker build`.
 
 Sidero Labs maintains a [repository of overlays](https://github.com/siderolabs/overlays).
+
+### Developing An Overlay
+
+Let's assume that you would like to contribute an overlay for a specific board, e.g. by contributing to the [`sbc-rockchip` repository](https://github.com/siderolabs/sbc-rockchip). Clone the repositry and insepct the existing overlays to understand the structure.
+
+Usually an overlay consist of a few key components:
+
+- `firmware`: contains the firmware files required for the board
+- `bootloader`: contains the bootloader, e.g. `u-boot` for the board
+- `dtb`: contains the device tree blobs for the board
+- `installer`: contains the installer that will be used to install this overlay on the node
+- `profile`: contains information
+
+1. For the new overlay, create any needed folders and `pkg.yaml` files.
+2. If your board introduces a new chipset that is not supported yet, make sure to add the firmware build for it.
+3. Add the necessary `u-boot` and `dtb` build steps to the `pkg.yaml` files.
+4. Proceed to add an installer, which is a small go binary that will be used to install the overlay on the node. Here you need to add the go `src/` as well as the `pkg.yaml` file.
+5. Lastly, add the profile information in the `profiles` folder.
+
+You are now ready to attempt building the overlay. It's recommend to push the build to a container registry to test the overlay with the Talos installer.
+
+```bash
+export USERNAME=octocat
+docker login ghcr.io
+make sbc-rockchip USERNAME=${USERNAME} PUSH=true
+```
+
+After building the overlay, take note of the pushed image tag, e.g. `664638a`, because you will need it for the next step. You can now build a flashable image using the command below.
+
+```bash
+export TALOS_VERSION=v1.7.6
+export USERNAME=octocat
+export BOARD=nanopi-r5s
+export CHIPSET=rk3568
+export TAG=664638a
+
+docker run --rm -t -v ./_out:/out -v /dev:/dev --privileged ghcr.io/siderolabs/imager:${TALOS_VERSION} \
+    metal --arch arm64 \
+    --base-installer-image="ghcr.io/siderolabs/installer:${TALOS_VERSION}" \
+    --overlay-name="${BOARD}" \
+    --overlay-image="ghcr.io/${USERNAME}/sbc-rockchip:${TAG}" \
+    --overlay-option="chipset=${CHIPSET}"
```

Comment:
this also is unused , i'll fix it when signing the commit

Reply:
or i'll add a separate section on using `overlay-option` since it's also supports loading from a file